Automattic/wp-calypso
# Gutenboarding: Update create-site step to redirect to IntentGathering if not creating a site

This PR fixes #41773 (prevents a user from being stuck in the `create-site` step), and in the process makes the following changes:

#### Changes proposed in this Pull Request

* Move `isRedirecting` to the onboard store so that the Header isn't managing that state (and so that it can be used to determine whether or not to render the `create-site` step)
* Only render the `create-site` step if a site is in the process of being created, otherwise redirect to the last step a user was on.
* Because of that, move the query string `?new` behaviour out of the `create-site` step, and have it be handled by the `OnboardingEdit` component, where most of the other redirect behaviour lives.
* Add a `newSite` check to the method used for creating a new site when a new user is created — this removes a bug where two sites were being created if a user created a site at the same time as creating a new user

#### Testing instructions



##### Regression testing:

From `/new`:

1. Create a site _and_ a new user at the same time
2. While logged in, create a new site
3. Create a new site, and from the editor, click Back (you should arrive at the beginning of `/new`)
4. While logged in, complete the new site flow up until the style step. Enter `/new/create-site?new` and ensure a new site is created.

##### Testing the fix for #41773 

1. From a logged-out session, go to `/new` and be sure to select a paid domain in the process
2. Create a new user account and complete signup, and you should be taken to the prelaunch `plans` step
3. Click back, and you should arrive at `/new`

And a question — is it a problem that all routes within `/new` now respond to the `?new` query parameter, so that the behaviour isn't dependent on the `create-site` step?

Fixes #41773
